File: ClientApp/src/dto/budgetItems/budgetItemCreateDto.ts
import { BudgetItemOperationType } from '../../enums/Budgets/budgetItemOperationType';
import { BudgetItemRegularityType } from '../../enums/Budgets/budgetItemRegularityType';

export interface BudgetItemCreateDto {
  budgetId?: number;
  itemTitle: string;
  itemDescription: string;
  budgetItemOperationType: BudgetItemOperationType;
  budgetItemRegularityType: BudgetItemRegularityType;
  operationCost: string;
  operationDate: Date;
  tagIds: number[];
}

File: GlobalHub/Services/BudgetService/BudgetsService.Business/Dto/BudgetItems/BudgetItemCreateDto.cs
namespace BudgetsService.Business.Dto.BudgetItems;

public record BudgetItemCreateDto
{
    public string ItemTitle { get; init; }

    public string? ItemDescription { get; init; }

    public BudgetItemOperationType BudgetItemOperationType { get; init; }

    public BudgetItemRegularityType BudgetItemRegularityType { get; init; }

    public decimal OperationCost { get; init; }

    public DateTime OperationDate { get; init; }

    public IEnumerable<long> TagIds { get; init; } = new List<long>();

    public long BudgetId { get; init; }
}
